component/metrics/scraper/appendable.go

```diff
@@ -0,0 +1,82 @@
+package scraper
+
+import (
+	"context"
+	"sync"
+
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/agent/component/metrics"
+	"github.com/prometheus/prometheus/model/exemplar"
+	"github.com/prometheus/prometheus/storage"
+)
+
+type scrapeAppendable struct {
+	mut       sync.Mutex
+	buffer    map[int64][]metrics.FlowMetric
+	receivers []*metrics.Receiver
+}
+
+func newScrapeAppendable(receiver []*metrics.Receiver) *scrapeAppendable {
+	return &scrapeAppendable{
+		buffer:    make(map[int64][]metrics.FlowMetric),
+		receivers: receiver,
+	}
+}
+
+func (s *scrapeAppendable) Append(ref storage.SeriesRef, l labels.Labels, t int64, v float64) (storage.SeriesRef, error) {
+	// BIG TODO is that we probably want to move refid creation and caching into a shared cache between wal and scraper at some point.
+	// 	in the below the refid is never cached by the scrape pool which makes it sad. Its likely this / refid caching should
+	//  should create the refid cache
+	s.mut.Lock()
+	defer s.mut.Unlock()
```

Comment:
Maybe we dont need the mutexes? I dont think appendables allow multiple writes from different sources to appendable.